platform/kube/client_test.go

```diff
@@ -121,3 +123,16 @@ func deleteNamespace(cl kubernetes.Interface, ns string) {
 		glog.Infof("Deleted namespace %s", ns)
 	}
 }
+
+func TestPostFailsOnInvalidNamespace(t *testing.T) {
```

Comment:
I think we can leave client_test and mock config unmodified. The fact that k8s allows access to non-existent namespaces is a bug, and we don't need to hardcode that. Basically, if namespace doesn't exist, the underlying kube client should be throwing an error without any special handling by us.